### ISSUE TITLE
[BreadcrumbWithOverflow]: hidden sizing items cause accessibility issue with id

### ISSUE THREAD

**Issue by joyful-these** [carbon-design-system/ibm-products#2449]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

In the [BreadcrumbWithOverflow](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/main/packages/cloud-cognitive/src/components/BreadcrumbWithOverflow/BreadcrumbWithOverflow.js) component, the useEffect labeled "create hidden sizing items" contains hidden breadcrumbs that seems to use the same id as the displayed breadcrumb items, causing an accessibility issue. As seen in the screenshot below, the Accessibility Checker detects a violation with the breadcrumb id already being in use. The code in the inspector shows the hidden breadcrumbs above and the displayed breadcrumbs below.

<img width="1440" alt="Screen Shot 2022-11-08 at 4 37 27 PM" src="https://user-images.githubusercontent.com/108295261/200682672-1cd1d512-07fa-48a2-b76e-fd2acf51bc75.png">

> What browser are you working in?

Firefox

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

"@carbon/ibm-products": "^1.35.1"

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

akora-content-policies (the UI content bundle for the Hybrid DevOps and Management policy UIs)
Our current release has all fixes due by Nov 23

**Comment by matthewgallo** [comment]
Hey @joyful-these, can you provide an example of how you're using this part of the `PageHeader` (specifically the data you're passing to the `breadcrumb` prop)? From what I'm seeing, if the `id` is provided in the breadcrumb data it is added to the `li` element, not the `a` tag.

**Comment by matthewgallo** [comment]
Ah ok, I see. Let us know if you have any more questions!
